### DISCUSSION TITLE
What should be the ideal value of quartz.threadPool.threadCount?

### DISCUSSION THREAD

**Discussion by viveknuna** [quartznet/quartznet#2169] [Q&A]
What should be the ideal value of quartz.threadPool.threadCount? In the current system, We have set it to 18.

The below line gives the value as 8 in my case.

`Logger.LogInfo("Number Of Logical Processors: {0}", Environment.ProcessorCount);`

My service is running on one POD.

So Can I increase it more? If yes then what value shall I have?

**Comment by lahma** [comment]
As always, I'm afraid it depends. The limit allows you to reduce the possibility to exhaust system resources, but it can vary a lot by job behavior. 

For example if your jobs are not CPU bound and external I/O bound, you could have much higher value as resource consumption happens outside - say in external API service.

Quartz also utilizes tasks by default as of v3 instead of dedicated threads which allows more concurrency and utilizes runtime's task processing pool. If you never have more than 10 jobs running concurrently then the tweaking doesn't matter much.

You need to measure and find the sweet spot 🙂
